contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,280 @@
+        int256 longAmount = _vault.longAmounts.length > 0 ? FixedPointInt256.uintToInt(_vault.longAmounts[0]) : 0;
+
+        bool expired = now > OtokenInterface(short).expiryTimestamp();
+        bool isPut = OtokenInterface(short).isPut();
+
+        if (!expired) {
+            int256 shortStrike = FixedPointInt256.uintToInt(OtokenInterface(short).strikePrice());
+            int256 longStrike = 0;
+            if (_vault.longOtokens.length > 0)
+                longStrike = FixedPointInt256.uintToInt(OtokenInterface(_vault.longOtokens[0]).strikePrice());
+
+            if (isPut) {
+                netOtoken = _calculateNonExpiredPutNetValue(shortAmount, longAmount, shortStrike, longStrike);
+            } else {
+                netOtoken = _calculateNonExpiredCallNetValue(shortAmount, longAmount, shortStrike, longStrike);
+            }
+        } else {
+            int256 shortCashValue = FixedPointInt256.uintToInt(getExpiredCashValue(short));
+            int256 longCashValue = 0;
+            if (_vault.longOtokens.length > 0)
+                longCashValue = FixedPointInt256.uintToInt(getExpiredCashValue(_vault.longOtokens[0]));
+            if (isPut) {
+                netOtoken = _calculateExpiredPutNetValue(shortAmount, longAmount, shortCashValue, longCashValue);
+            } else {
+                (uint256 underlyingPrice, ) = _getUnderlyingPrice(address(short));
+                int256 underlyingPriceInt = FixedPointInt256.uintToInt(underlyingPrice);
+                netOtoken = _calculateExpiredCallNetValue(
+                    shortAmount,
+                    longAmount,
+                    shortCashValue,
+                    longCashValue,
+                    underlyingPriceInt
+                );
+            }
+        }
+    }
+
+    /**
+     * @dev calculate non-expired net put spread value.
+     *
+     * Formula: net = (long strike * min (short amount, long amount)) - (short amount * short strike)
+     *
+     * @return net value
+     */
+    function _calculateNonExpiredPutNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortStrike,
+        int256 _longStrike
+    ) internal pure returns (int256) {
+        return _longStrike.mul(FixedPointInt256.min(_shortAmount, _longAmount)).sub(_shortAmount.mul(_shortStrike));
+    }
+
+    /**
+     * @dev calculate non-expired net call spread value.
+     *                                                min (long amount, short amount) * max (0, long strike - short strike)
+     * net = min(0, long amount - short amount) -  --------------------------------------------------------------------------
+     *                                                                             long strike
+     *
+     * @dev if long strike = 0 (no long token), then return net = short amount.
+     * @return net value
+     */
+    function _calculateNonExpiredCallNetValue(
```

Comment:
same comment as before

Reply:
also does it make a difference if expired or not?

Reply:
for calls, the formula is different